core/include/thread.h

```diff
@@ -11,6 +11,100 @@
+ *       the number of threads in application is significantly smaller than the
+ *       number of available priorities, so assigning distinct priorities per
+ *       thread should not be a problem. Only assign the same priority to
+ *       multiple threads if you know what you are doing!
+ *
+ * Thread Behavior
+ * ===============
+ * In addition to the priority, flags can be used when creating a thread to
+ * alter the thread's behavior after creation. The following flags are available:
+ *
+ *  @ref THREAD_CREATE_SLEEPING :   the newly created thread will be put to sleeping
+ *                                  state and must be woken up manually
+ *
+ *  @ref THREAD_CREATE_WOUT_YIELD : the newly created thread will not run
+ *                                  immediately after creation
+ *
+ *  @ref THREAD_CREATE_STACKTEST :  write markers into the thread's stack to measure
+ *                                  the stack's memory usage (for debugging and
+ *                                  profiling purposes)
+ *
+ * Thread creation
+ * ===============
+ * Creating a new thread is internally done in two steps:
+ * 1. the new thread's stack is initialized depending on the platform
+ * 2. the new thread is added to the scheduler and the scheduler is run (if not
+ *    indicated otherwise)
+ *
+ * @note Creating threads from within an ISR is currently supported, however it
+ *       is considered to be a bad programming practice and we strongly
+ *       discourage you from doing so.
+ *
+ * Usage
+ * -----
+ * ~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * char rcv_thread_stack[THREAD_STACKSIZE_MAIN];
+ *
+ * void *rcv_thread(void *arg)
+ * {
+ *     msg_t m;
+ *
+ *     while (1) {
+ *         msg_receive(&m);
+ *         printf("Got msg from %" PRIkernel_pid "\n", m.sender_pid);
+ *     }
+ *
+ *     return NULL;
+ * }
+ *
+ * int main(void)
+ * {
+ *     thread_create(rcv_thread_stack, sizeof(rcv_thread_stack),
+ *                   THREAD_PRIORITY_MAIN - 1, THREAD_CREATE_STACKTEST,
+ *                   rcv_thread, NULL, "rcv_thread");
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Reading from the top down, you can see that first, stack memory for our thread
+ * `rcv_thread` is preallocated, followed by an implementation of the thread's
+ * tasks. Communication between threads is done using @ref core_msg: in this case,
+ * rcv_thread` will print the process id of each other thread that sent a message
+ * to `rcv_thread.
+ *
+ * After it has been properly defined, `rcv_thread` is created with a call to
+ * @ref thread_create() in `main()`. It is assigned a priority of
+ * `THREAD_PRIORITY_MAIN - 1`, i.e. a slightly *higher* priority than the main
+ * thread. Since neither the `THREAD_CREATE_SLEEPING` nor the
+ * `THREAD_CREATE_WOUT_YIELD` flag is set, `rcv_thread` will be executed
+ * immediately.
+ *
+ * @note To prevent messages from being dropped when they can't be handled right
```

Comment:
Ah, true. I just wanted to make sure people don't forget to initialize it in case they need it, as this is a mistake that is easy to make and might take a while to catch. Imo, the details should be defined in the messaging documentation, so how about...
```
 * @note If the messages to the thread are sent using @ref msg_try_send() or
 *       from an ISR, activate your thread's message queue by calling
 *       @ref msg_init_queue() to prevent messages from being dropped when
 *       they can't be handled right away. The same applies if you'd like
 *       msg_send() to your thread to be non-blocking. For more details, see
 *       @ref core_msg "the Messaging documentation".
```
?